### ISSUE TITLE
'cannot prefetch a directory' crash

### ISSUE THREAD

**Issue by keith** [bazelbuild/bazel#17186]
### Description of the bug:

Using HEAD bazel and build without the bytes (I assume that's related), we've started seeing this crash.

```
(12:11:39) FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.RuntimeException: Unrecoverable error while evaluating node 'ActionLookupData{actionLookupKey=ConfiguredTargetKey{label=//Modules/ResourceSupport:ResourceSupportTests.__internal__.__test_bundle, config=BuildConfigurationKey[f8cc4d30b745f358f412b4c4b1f79432498fe02f6a0c5d873539031775299854]}, actionIndex=10}' (requested by nodes 'File:[[<execution_root>]bazel-out/applebin_ios-ios_x86_64-fastbuild-ST-5f721879e0e1/bin]Modules/ResourceSupport/ResourceSupportTests.xctest')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:680)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:382)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: cannot prefetch a directory
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
	at com.google.devtools.build.lib.remote.AbstractActionInputPrefetcher.prefetchFiles(AbstractActionInputPrefetcher.java:186)
	at com.google.devtools.build.lib.remote.AbstractActionInputPrefetcher.prefetchFiles(AbstractActionInputPrefetcher.java:176)
	at com.google.devtools.build.lib.exec.AbstractSpawnStrategy$SpawnExecutionContextImpl.prefetchInputs(AbstractSpawnStrategy.java:248)
	at com.google.devtools.build.lib.exec.SpawnRunner$SpawnExecutionContext.prefetchInputsAndWait(SpawnRunner.java:165)
	at com.google.devtools.build.lib.exec.local.LocalSpawnRunner.exec(LocalSpawnRunner.java:145)
	at com.google.devtools.build.lib.exec.AbstractSpawnStrategy.exec(AbstractSpawnStrategy.java:152)
	at com.google.devtools.build.lib.exec.AbstractSpawnStrategy.exec(AbstractSpawnStrategy.java:112)
	at com.google.devtools.build.lib.exec.SpawnStrategyResolver.exec(SpawnStrategyResolver.java:45)
	at com.google.devtools.build.lib.analysis.actions.SpawnAction.execute(SpawnAction.java:342)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.executeAction(SkyframeActionExecutor.java:1082)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.run(SkyframeActionExecutor.java:1003)
	at com.google.devtools.build.lib.skyframe.ActionExecutionState.runStateMachine(ActionExecutionState.java:160)
	at com.google.devtools.build.lib.skyframe.ActionExecutionState.getResultOrDependOnFuture(ActionExecutionState.java:93)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor.executeAction(SkyframeActionExecutor.java:516)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.checkCacheAndExecuteIfNeeded(ActionExecutionFunction.java:827)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.computeInternal(ActionExecutionFunction.java:323)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.compute(ActionExecutionFunction.java:161)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:591)
	... 4 more
```

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

- First known bad commit: 9860051afe7ccf0902f1ce3dbbc9066088d7c849
- Last known good commit: e7be4b1bcda6f59eaecb4d0a46fdefca5c45ccf4

Diff: https://github.com/bazelbuild/bazel/compare/e7be4b1bcda6f59eaecb4d0a46fdefca5c45ccf4...9860051afe7ccf0902f1ce3dbbc9066088d7c849

Culprit https://github.com/bazelbuild/bazel/commit/763f966af5ebc6d3bdf97feb150ebace6a2cebd2

@tjgq what is the suggested workaround in this case? I suppose we've been using this for a while successfully

### Which operating system are you running Bazel on?

macOS

### What is the output of `bazel info release`?

9860051afe7ccf0902f1ce3dbbc9066088d7c849

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.

last_green with bazelisk.

**Comment by brentleyjones** [comment]
Probably fixed by https://github.com/bazelbuild/bazel/pull/17183?

**Comment by keith** [comment]
ah probably, nice timing, i'll verify after it lands

**Comment by tjgq** [comment]
Sorry about that! As @brentleyjones said, it should have been fixed by 4c57def. Please reopen otherwise.
